doc/design/cluster_train/PServer_design_doc.md

```diff
@@ -0,0 +1,261 @@
+# Design Doc: Parameter Server Process
+
+Parameter Server process 是Paddle中负责模型的存储，更新和模型分片一致性的组件，在整个系统中的作用请参考 [distributed training design doc](./README.md) ，本文档包含PServer，PClient，PServerContoller等，涉及到的配置参数均使用大写字母
+
+<img src="src/paddle-model-sharding.png" width="500"/>
+
+## 术语
+
+- PServer: Parameter Server Server，负责模型存储，调用分布式更新，响应PClient请求
+- PClient: Parameter Server Client，负责均衡PServer请求，打包并转发RPC请求
+- PServerController：负责启动Server，动态扩容，容灾等
+- model: 指深度学习训练之后得到的所有参数，使用这个神经网络可以完成对新数据的预测
+- Tensor: 一个NDArray结构，Trainer与PServer, PClient交互的基本数据结构
+- shard: 全量模型在某个PServer上的局部分片，通常指将一个Model整体拆分成多份的其中的一份。
+- parameter block: 多个parameter block构成一个model shard(现存的model并行策略是parameter block based，在新架构中继续沿用)
+
+##  PServer
+
+PServer负责以下功能:
+
+1、模型存储，2、注册服务并监听端口事件，3、PServer个数的动态扩张收缩，4、负责序列化传输数据。
```

Comment:
Do we need to support the varying number of parameter server instances in a distributed training job? I don't think that we have to. And I think it would be extremely hard to re-sharding the model parameters when the number changes.